src/settings/language-modal/LanguageModal.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { translate } from 'react-i18next'
+import i18n, { localesList } from '../../i18n'
+import { getLanguage } from '../../lib/i18n'
+
+// Components
+import { Modal, ModalBody, ModalActions } from '../../components/modal/Modal'
+import SpeakerIcon from '../../icons/StrokeSpeaker'
+import Button from '../../components/button/Button'
+
+const LanguageModal = ({ t, tReady, onLeave, link, className, ...props }) => {
+  const handleClick = (lang) => {
+    console.log(lang)
+    i18n.changeLanguage(lang)
+    onLeave()
+  }
+
+  return (
+    <Modal {...props} className={className} onCancel={onLeave} >
+      <ModalBody icon={SpeakerIcon}>
+        <p className='gray w-80 center'>{t('languageModal.description')}</p>
+        <div className='pa2 flex flex-wrap'>
+          { localesList.map((lang) =>
+            <button
```

Comment:
For some reason the button styles need to include a background color to reset them on FF, and we need to give the text more space. Perhaps the modal could be wider.

<img width="872" alt="screenshot 2018-10-31 at 11 12 35" src="https://user-images.githubusercontent.com/58871/47784566-e4e74b00-dcfd-11e8-82f9-4930f9edb3b0.png">